packages/utils/src/json-schema.ts

```diff
@@ -1,24 +1,253 @@
-// Note - @types/json-schema@7.0.4 added some function declarations to the type package
-// If we do export *, then it will also export these function declarations.
-// This will cause typescript to not scrub the require from the build, breaking anyone who doesn't have it as a dependency
-
-// eslint-disable-next-line import/no-extraneous-dependencies
-export {
-  JSONSchema4,
-  JSONSchema4Type,
-  JSONSchema4TypeName,
-  JSONSchema4Version,
-  JSONSchema6,
-  JSONSchema6Definition,
-  JSONSchema6Type,
-  JSONSchema6TypeName,
-  JSONSchema6Version,
-  JSONSchema7,
-  JSONSchema7Array,
-  JSONSchema7Definition,
-  JSONSchema7Type,
-  JSONSchema7TypeName,
-  JSONSchema7Version,
-  ValidationError,
-  ValidationResult,
-} from 'json-schema';
+/**
+ * This is a fork of https://github.com/DefinitelyTyped/DefinitelyTyped/blob/13f63c2eb8d7479caf01ab8d72f9e3683368a8f5/types/json-schema/index.d.ts
+ * We intentionally fork this because:
+ * - ESLint ***ONLY*** supports JSONSchema v4
+ * - We want to provide stricter types
+ */
+
+//==================================================================================================
+// JSON Schema Draft 04
+//==================================================================================================
+
+/**
+ * @see https://tools.ietf.org/html/draft-zyp-json-schema-03#section-5.1
+ */
+export type JSONSchema4TypeName =
+  | 'string' //
```

Comment:
[Nitpick] Here and elsewhere

```suggestion
  | 'string'
```